distribution/docker/src/docker/Dockerfile

```diff
@@ -259,6 +261,27 @@ RUN sed -i -e 's/ES_DISTRIBUTION_TYPE=tar/ES_DISTRIBUTION_TYPE=docker/' bin/elas
     chmod 0775 bin config config/jvm.options.d data logs plugins && \\
     find config -type f -exec chmod 0664 {} +
 
+<% if (docker_base == "cloud") { %>
+# Preinstall common plugins
+COPY repository-s3-${version}.zip repository-gcs-${version}.zip repository-azure-${version}.zip /tmp/
+RUN bin/elasticsearch-plugin install --batch \\
+    file:/tmp/repository-s3-${version}.zip \\
+    file:/tmp/repository-gcs-${version}.zip \\
+    file:/tmp/repository-azure-${version}.zip
+
+# I tried to use `ADD` here, but I couldn't force it to do what I wanted
+COPY filebeat-${version}.tar.gz metricbeat-${version}.tar.gz /tmp/
+RUN mkdir -p /opt/filebeat /opt/metricbeat && \\
+    tar xf /tmp/filebeat-${version}.tar.gz -C /opt/filebeat --strip-components=1 && \\
+    tar xf /tmp/metricbeat-${version}.tar.gz -C /opt/metricbeat --strip-components=1
```

Comment:
@alpar-t @mieciu I'd still like to know what Cloud actually needs from the Beats distributions. I assume the entire distro isn't required. This isn't essential now, but anything we can trim from the image will help to keep the size down.